marketplace-infrastructure/src/event_webhook/mod.rs

```
@@ -0,0 +1,48 @@
+#[cfg(test)]
+mod tests;
+
+use async_trait::async_trait;
+use log::{error, info};
+use marketplace_domain::{Event, EventListener};
+use url::Url;
+
+const WEBHOOK_TARGET_ENV_VAR: &str = "EVENT_WEBHOOK_TARGET";
+
+pub struct EventWebHook {
+	web_client: reqwest::Client,
+}
+
+impl EventWebHook {
+	pub fn new(client: reqwest::Client) -> Self {
+		Self { web_client: client }
+	}
+}
+
+#[async_trait]
+impl EventListener for EventWebHook {
+	async fn on_event(&self, event: &Event) {
```

I would rather refactor this into a faillible function so that we can leverage the `?` syntax.
Then log and ignore the error in the `on_event`
This would limit the nesting

---

Not sure about that. 
Listeners are by definition infallible.
You put stuff in it, maybe things happen, maybe not. You don't care and continue your main execution.

---

done